docs/docs/components/guidelines/README.md

```diff
@@ -552,4 +762,823 @@ end user. Generate and use a GUID for the shared secret value, save it to a
+limits (or at least be able to gracefully handle them), making the components
+robust and ensuring that no events are missed.
+
+As a rule of thumb, when an API endpoint is paginated verify that its
+encapsulating method:
+
+1. Is defined as an `async` generator that yields a page of records on each
+   iteration
+2. The generator also accepts an optional token for the next page to retrieve
+   via the API, and also yields a token for the page following the one queried
+   last. A variation of this is allowing the generator to accept a record ID
+   representing the last successfully processed record.
+3. Process the records as soon as they are yielded by the generator, and avoid
+   accumulating records (e.g. in a collection/array) before processing them if
+   possible. If the API records need to be processed as a collection (e.g. they
+   need to be sorted), check first if such processing can be done by the API.
+4. After processing each record or page, persist the record ID or next page
+   token, respectively, so that the component keeps track of the progress made
+   (like a checkpoint)
+
+Below is an example of a paginated API call wrapped around an `async` generator
+method, and a component making use of it:
+
+- `components/cloud_storage/cloud_storage.app.js`
+
+  ```javascript
+  const axios = require("axios");
+
+  module.exports = {
+    type: "app",
+    methods: {
+      /**
+      * This method yields pages of files present in the drive with `driveId`
+      * under the path `path` (the root of the drive if unspecified)
+      *
+      * @param {string} driveId the ID of the cloud storage drive
+      * @param {object} opts additional options
+      * @param {string} [opts.path="/"] the path in the drive from where to
+      * retrieve the list of files
+      * @param {string} [opts.lastFileId] the ID of file from which to list
+      * files, exclusive: only files created after this one will be retrieved
+      * @yields {object[]} the list of files in the drive, under the specified
+      * path
+      */
+      async *listFiles(driveId, {
+        path = "/",
+        lastFileId,
+      }) {
+        const url = this._listFilesUrl();
+        const params = {
+          drive_id: driveId,
+          path,
+          since_id: lastFileId,
+        };
+        const requestConfig = {
+          ...this._makeRequestConfig(),
+          params,
+        };
+        while (true) {
+          const { data } = await axios.get(url, requestConfig);
+          const { records: files = [] } = data;
+          if (files.length <= 0) {
+            // No more files to scan, we can complete the pagination scan now
+            return;
+          }
+
+          yield files;
+        }
+      },
+    },
+  };
+  ```
+
+  ::: tip
+  Note that the interface around this API call only requires the drive ID, and
+  hides the details regarding the pagination details. Any additional parameter
```

Comment:
"hides the details regarding the pagination details" -> "hides the details regarding pagination"